app/controllers/logs_controller.rb

```diff
@@ -18,6 +18,7 @@ def index
               current_organisation.ip_addresses.intersection([log_search_form.ip])
             end
       logs = Gateways::Sessions.search(ips:, success:)
+      @current_location = Ip.find_by(address: log_search_form.ip).location if logs.present?
```

Comment:
You could user the `ips` variable here instead of `log_search_form.ip` Just so that it is 100% impossible to get information on a location a regular admin user does not control.